packages/interactor/src/locator.ts

```diff
@@ -0,0 +1,21 @@
+export type LocatorFn<E extends Element> = (element: E) => string;
+
+export type LocatorSpecification<E extends Element> = Record<string, LocatorFn<E>>;
+
+export class Locator<E extends Element> {
+  constructor(public locatorFn: LocatorFn<E>, public value: string, public name?: string) {}
+
+  get description(): string {
+    if(this.name) {
+      let name = this.name.toString().replace(/^by/, '');
+      name = name.charAt(0).toLowerCase() + name.slice(1);
+      return `with ${name} ${JSON.stringify(this.value)}`;
```

Comment:
Why do we need to do `JSON.stringify()` if `this.value` is of type `string`?

Reply:
Mostly I prefer doing `JSON.stringify(foo)` over simply `"${foo}"` because it properly handles stuff like double quotes contained in the string. It's not a huge deal, but IMO it's just a little bit nicer.